### PR TITLE
[INLONG-10193][Sort] Postgres connector support audit ID

### PR DESCRIPTION
**Pull request by XiaoYou201**
### Prepare a Pull Request
- [INLONG-10193][Sort] Postgres connectors support audit ID

- Fixes https://github.com/apache/inlong/issues/10193

- parent issue: https://github.com/apache/inlong/issues/8803

### Motivation
Postgres connector not support audit function. So this pr will make it support audit function.

### Modifications
Modify Postgres connector source.when source receive data, it will send audit information at same time.Some implement are as follows:

Program will get the audit information from the incoming sql statement and pass it to the deserialization classes. During deserialization, the program will report the number of data received.

企业微信截图_e874eff6-387b-4dda-abd4-201fc3c5b3a2

### CODE REVIEW

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/postgre/RowDataDebeziumDeserializeSchema.java [hunk 1]
```diff
@@ -0,0 +1,684 @@
...
+        this.appendMetadataCollector = new AppendMetadataCollector(metadataConverters);
+        this.physicalConverter =
+                createConverter(
+                        checkNotNull(physicalDataType),
+                        serverTimeZone,
+                        userDefinedConverterFactory);
+        this.resultTypeInfo = checkNotNull(resultTypeInfo);
+        this.validator = checkNotNull(validator);
+        this.changelogMode = checkNotNull(changelogMode);
+        this.sourceMetricData = sourceMetricData;
```
Comment (aloyszhang) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/postgre/RowDataDebeziumDeserializeSchema.java:132]: Where's the place init the `RowDataDebeziumDeserializeSchema` with `SourceMetricData`

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/postgre/PostgresValueValidator.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+package org.apache.inlong.sort.postgre;
+
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.types.RowKind;
+
+/** The {@link RowDataDebeziumDeserializeSchema.ValueValidator} for Postgres connector.
+ *
+ * Copy from com.ververica:flink-connector-postgres-cdc-2.3.0
+ * */
+public final class PostgresValueValidator
```
Comment (aloyszhang) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/postgre/PostgresValueValidator.java:27]: Any differences with the original class in flink-connector?

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/postgre/RowDataDebeziumDeserializeSchema.java [hunk 3]
```diff
@@ -0,0 +1,684 @@
...
+                DeserializationRuntimeConverterFactory userDefinedConverterFactory) {
+            this.userDefinedConverterFactory = userDefinedConverterFactory;
+            return this;
+        }
+
+        public Builder setChangelogMode(DebeziumChangelogMode changelogMode) {
+            this.changelogMode = changelogMode;
+            return this;
+        }
+        public Builder setSourceMetricData(MetricOption metricOption) {
```
Comment (EMsnap) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/postgre/RowDataDebeziumDeserializeSchema.java:247]: this method is never used and metrics won't be reported

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/postgre/RowDataDebeziumDeserializeSchema.java (hunk 1 above)
Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/postgre/RowDataDebeziumDeserializeSchema.java:132]: Through builder.builder() in RowDataDebeziumDeserializeSchema

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/postgre/PostgresValueValidator.java (hunk 2 above)
Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/postgre/PostgresValueValidator.java:27]: PostgresValueValidator implements RowDataDebeziumDeserializeSchema.ValueValidator, but I modify RowDataDebeziumDeserializeSchema, so, setValueValidator(ValueValidator validator) need validator which implements modified RowDataDebeziumDeserializeSchema.ValueValidator. ~

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/postgre/RowDataDebeziumDeserializeSchema.java (hunk 3 above)
Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/postgre/RowDataDebeziumDeserializeSchema.java:247]: It used in PostgreSQLTableSource to init DebeziumDeserializationSchema.

DebeziumDeserializationSchema<RowData> deserializer = RowDataDebeziumDeserializeSchema.newBuilder()....setSourceMetricData(metricOption).build()

Comment (EMsnap) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/postgre/RowDataDebeziumDeserializeSchema.java:247]: setSourceMetricData(MetricOption metricOption) the parameter name doesn't match the method name

Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/postgre/RowDataDebeziumDeserializeSchema.java:247]: fix it～

Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/postgre/RowDataDebeziumDeserializeSchema.java]: fix it

Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/postgre/RowDataDebeziumDeserializeSchema.java]: fix it

**Review by aloyszhang — DISMISSED** [review]

**Review by EMsnap — DISMISSED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**